app/controllers/campuses_controller.ts

```diff
@@ -0,0 +1,38 @@
+import type { HttpContext } from "@adonisjs/core/http";
+
+import Campus from "#models/campus";
+import { showValidator } from "#validators/show";
+
+export default class CampusesController {
+  /**
+   * Display a list of resource
+   */
+  async index({ request }: HttpContext) {
+    const campuses = await Campus.query().withScopes((scopes) => {
+      scopes.handleSearchQuery(
+        request.only(["id", "name", "createdAt", "updatedAt"]),
+      );
+      scopes.includeRelations(request.only(["buildings"]));
+      scopes.handleSortQuery(request.input("sort"));
+    });
+
+    return campuses;
```

Comment:
nitpick: you can return campuses inline, instead of creating the variable for it